api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -3,65 +3,192 @@
+    assertThat(dbInstitution.getEmailDomains()).isEmpty();
+    assertThat(dbInstitution.getEmailAddresses()).isEmpty();
+
+    dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailDomains()).isEqualTo(emailDomains);
+    assertThat(dbInstitution.getEmailAddresses()).isEqualTo(emailAddresses);
   }
 
   @Test
   public void test_findOneByShortName() {
-    assertThat(institutionDao.findOneByShortName("Broad")).hasValue(testInst);
+    assertThat(institutionDao.findOneByShortName("Broad"))
+        .hasValue(institutionNoEmailAddressesOrDomains);
+    assertThat(institutionDao.findOneByShortName("Broad1"))
+        .hasValue(institutionWithEmailAddressesAndDomains);
     assertThat(institutionDao.findOneByShortName("Verily")).isEmpty();
+  }
+
+  @Test
+  public void test_updateAllNewEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses =
+        Sets.newHashSet(
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress1"),
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress2"));
+    institutionWithEmailAddressesAndDomains.setEmailAddresses(newEmailAddresses);
+    institutionDao.save(institutionWithEmailAddressesAndDomains);
+
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailAddresses()).containsExactlyElementsIn(newEmailAddresses);
+  }
+
+  @Test
+  public void test_updateWithExistingEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses =
+        Sets.newHashSet(
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress"),
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress2"));
+    institutionWithEmailAddressesAndDomains.setEmailAddresses(newEmailAddresses);
+    institutionDao.save(institutionWithEmailAddressesAndDomains);
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailAddresses()).containsExactlyElementsIn(newEmailAddresses);
+  }
 
-    DbInstitution otherInst = new DbInstitution("Verily", "An Alphabet Company");
-    otherInst = institutionDao.save(otherInst);
-    assertThat(institutionDao.findOneByShortName("Verily")).hasValue(otherInst);
+  @Test
+  public void test_updateRemoveAllEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses = Sets.newHashSet();
+    institutionWithEmailAddressesAndDomains.setEmailAddresses(newEmailAddresses);
+    institutionDao.save(institutionWithEmailAddressesAndDomains);
```

Comment:
please overwrite the arg passed to save().

Reply:
good catch